libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFAction.m

```diff
@@ -756,24 +756,42 @@ - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response er
     }
     
     // Surface error back if we run into JSON parsing error on a successful HTTP response
-    if (jsonParseError && error && requestSucceeded) {
+    if (jsonParseError && requestSucceeded) {
         NetworkWarn(@"Error while parsing response; it doesn't appear to be JSON");
         
-        *error = [NSError errorWithDomain:CSFNetworkErrorDomain
-                                     code:CSFNetworkJSONInvalidError
-                                 userInfo:@{ NSLocalizedDescriptionKey: @"Processing response content failed",
-                                             NSUnderlyingErrorKey: jsonParseError,
-                                             CSFNetworkErrorActionKey: self }];
+        if (error) {
```

Comment:
Logic's the same, but was hard to read in its previous context.  Also missed the warning if `error` was `nil`.